src/hooks/usePodOwnerProfile/index.js

```diff
@@ -21,21 +21,40 @@
 
 import { useState, useEffect } from "react";
 import { useRouter } from "next/router";
+import { useSession } from "@inrupt/solid-ui-react";
+import { getResourceInfo, getPodOwner } from "@inrupt/solid-client";
 import { joinPath } from "../../stringHelpers";
 import useAuthenticatedProfile from "../useAuthenticatedProfile";
 import usePodRoot from "../usePodRoot";
 import useFetchProfile from "../useFetchProfile";
 
 export default function usePodOwnerProfile() {
+  const { session } = useSession();
+  const { fetch } = session;
   const [profile, setProfile] = useState();
   const [error, setError] = useState();
   const router = useRouter();
   const decodedResourceUri = decodeURIComponent(router.query.iri);
+  const [podOwnerWebId, setPodOwnerWebId] = useState(null);
   const podRoot = usePodRoot(decodedResourceUri, null);
-  const profileIri = podRoot && joinPath(podRoot, "profile/card#me"); // we won't need to do this once ownership is available
+  const profileIri =
+    podOwnerWebId || (podRoot && joinPath(podRoot, "profile/card#me")); // we won't need to do this once ownership is available
   const { data: authProfile, error: authError } = useAuthenticatedProfile();
   const { data: ownerProfile, error: ownerError } = useFetchProfile(profileIri);
 
+  useEffect(() => {
```

Comment:
Let's create this into a hook instead, makes it easier to reuse the functionality.